File: docs/code-examples/all/series_point_style.py
"""Log a scalar over time."""

from math import cos, sin, tau

import rerun as rr

rr.init("rerun_example_series_point_styling", spawn=True)

# Set up plot styling:
# They are logged timeless as they don't change over time and apply to all timelines.
# Log two lines series under a shared root so that they show in the same plot by default.
rr.log(
    "trig/sin",
    rr.SeriesPoint(
        color=[255, 0, 0],
        name="sin(0.01t)",
        marker="circle",
        marker_size=4,
    ),
    timeless=True,
)
rr.log(
    "trig/cos",
    rr.SeriesPoint(
        color=[0, 255, 0],
        name="cos(0.01t)",
        marker="cross",
        marker_size=2,
    ),
    timeless=True,
)

# Log the data on a timeline called "step".
for t in range(0, int(tau * 2 * 10.0)):
    rr.set_time_sequence("step", t)

    rr.log("trig/sin", rr.Scalar(sin(float(t) / 10.0)))
    rr.log("trig/cos", rr.Scalar(cos(float(t) / 10.0)))

File: docs/code-examples/all/series_point_style.cpp
// Log a scalar over time.

#include <rerun.hpp>

#include <cmath>

constexpr float TAU = 6.28318530717958647692528676655900577f;

int main() {
    const auto rec = rerun::RecordingStream("rerun_example_series_line_styling");
    rec.spawn().exit_on_failure();

    // Set up plot styling:
    // They are logged timeless as they don't change over time and apply to all timelines.
    // Log two point series under a shared root so that they show in the same plot by default.
    rec.log_timeless(
        "trig/sin",
        rerun::SeriesPoint()
            .with_color({255, 0, 0})
            .with_name("sin(0.01t)")
            .with_marker(rerun::components::MarkerShape::Circle)
            .with_marker_size(4)
    );
    rec.log_timeless(
        "trig/cos",
        rerun::SeriesPoint()
            .with_color({0, 255, 0})
            .with_name("cos(0.01t)")
            .with_marker(rerun::components::MarkerShape::Cross)
            .with_marker_size(2)
    );

    // Log the data on a timeline called "step".
    for (int t = 0; t < static_cast<int>(TAU * 2.0 * 10.0); ++t) {
        rec.set_time_sequence("step", t);

        rec.log("trig/sin", rerun::Scalar(sin(t / 10.0)));
        rec.log("trig/cos", rerun::Scalar(cos(static_cast<float>(t) / 10.0f)));
    }
}
